py_mmd_tools/nc_to_mmd.py

```diff
@@ -1598,35 +1599,54 @@ def to_mmd(
-            coverage.
-        geographic_extent_rectangle : dict, optional
-            The geographic extent of the datasets defined as a rectangle
-            in lat/lon projection. The extent is defined using the
-            following child elements: {
-                'geospatial_lat_max': geospatial_lat_max
-                    - The northernmost point covered by the dataset.
-                'geospatial_lat_min': geospatial_lat_min
-                    - The southernmost point covered by the dataset.
-                'geospatial_lon_min': geospatial_lon_min
-                    - The easternmost point covered by the dataset.
-                'geospatial_lon_max': geospatial_lon_max
-                    - The westernmost point covered by the dataset.
-            }
+        parent : str, optional
+            ID of parent dataset.
+        overrides : dict, optional
+            A dictionary with overrides for certain MMD fields:
+
+            time_coverage_start : str
+                The start date and time, in iso8601 format, for the
+                dataset coverage.
+            time_coverage_end : str
+                The end date and time, in iso8601 format, for the
+                dataset coverage.
+            geographic_extent_rectangle : dict
+                The geographic extent of the datasets defined as a
+                rectangle in lat/lon projection. The extent is defined
+                using the following child elements: {
+                    'geospatial_lat_max': geospatial_lat_max
+                        - The northernmost point covered by the
+                          dataset.
+                    'geospatial_lat_min': geospatial_lat_min
+                        - The southernmost point covered by the
+                          dataset.
+                    'geospatial_lon_min': geospatial_lon_min
+                        - The easternmost point covered by the
+                          dataset.
+                    'geospatial_lon_max': geospatial_lon_max
+                        - The westernmost point covered by the
+                          dataset.
+                }
+            dataset_citation : dict
+                An alternative dataset citation. This can be useful if
+                the citation refers to a parent dataset or a DOI.
+            platform : dict
+                A dictionary specifying the platform according to MMD
+                guidelines.
 
         This list can be extended but requires some new code...
         """
         if collection is not None and type(collection) is not str:
             raise ValueError("collection must be of type str")
 
-        # kwargs that were not added in the function def:
-        time_coverage_start = kwargs.pop("time_coverage_start", "")
-        time_coverage_end = kwargs.pop("time_coverage_end", "")
-        geographic_extent_rectangle = kwargs.pop("geographic_extent_rectangle", "")
+        # Overrides
+        if overrides is None:
```

Comment:
Why is this check needed? Can we not just set overrides default value to `{}`?

Reply:
See https://www.geeksforgeeks.org/default-arguments-in-python/ under "Using mutable objects as default argument values in python". This is why I generally use None as default for dicts, even when I don't expect the method to be called repeatedly.